student-projects/gregory/Sdutents-Stats/app.ts

```diff
@@ -0,0 +1,59 @@
+let StatType = 0;
+let Answer = "";
+while (!((StatType === 1) || (StatType === 2)))  {
+    Answer = prompt("Student Statistics. Do you want '1' - Best Grade or '2' - Worst Grade?");
+    if (Answer === null) {
+        alert("Looks like you don't want any grade. Good Day!");
+        throw "Canceled by the user.";
+    }
+    StatType = Number(Answer);
+    if (!((StatType === 1) || (StatType === 2))){
+        alert("ERROR - wrong input, it must be 1 or 2!");    
+    } 
+}
+
+let MinGrade = 100;
```

Comment:
It's better to initialize with a value that is out of range, like 101 or `Infinity`. Just to make sure all edge cases are handled.
It's possible to have a single variable for this task, we can initiate it with a ternary:
```typescript
let RequestedGrade = StatType === 1 ? 0 : 101;
```